ui/web/index.html

```diff
@@ -3,10 +3,307 @@
+                    </td>
+                    <td>
+                        <input type="button" value="Print" id="printGraphicButton">
+                    </td>
+                </tr>
+                <tr>
+                    <td>
+                        <div id="chartdiv" style="height:500px;width:300px; "></div>
+                    </td>
+                </tr>
+                </table>
+    </fieldset>
+
+    <script> 
+            var applicationContext = {}; 
+                                              
+        function loadTasks() {
+            var excelFile = $('#excelFile').get(0).files.item(0);
+            $.ajax({
+                type: 'PUT',
+                url: 'http://localhost:5000/api/excelWorkbook',
+                data: excelFile,
+                contentType: 'application/vnd.ms-excel',
+                processData: false,
+                success: function (data) {
+                    var requestBody = {
+                        sheet: $("#sheetName").val(),
+                        first_row: +$("#firstRow").val(),
+                        last_row: +$("#lastRow").val(),
+                        rows_to_skip: $("#rowSkip").val(),
+                        columns_mapping: {
+                            uid: $("#uid").val(),
+                            name: $("#nam").val(),
+                            blockers: $("#blockers").val(),
+                            min_estimate: $("#minEst").val(),
+                            normal_estimate: $("#norEst").val(),
+                            max_estimate: $("#maxEst").val()
+                        }
+                    };
+                    alert("Response received: " + JSON.stringify(data));
+                    requestBody.file = data.tempFileName;
+                    $.ajax({
+                        type: 'POST',
+                        url: 'http://localhost:5000/api/excelWorkbook/tasks',
+                        contentType: 'application/json',
+                        data: JSON.stringify(requestBody),
+                        success: function (data) {
+                            alert("Response received: " + JSON.stringify(data));                           
+                            applicationContext.tasks = data.tasks;    
+                            var selectList = document.createElement("select");
+                            selectList.setAttribute("id", "tasks");
+                            selectList.setAttribute("name", "tasks");
+                            choiceGraphic.appendChild(selectList);
+                            for (var i = 0; i < applicationContext.tasks.length; i++) {
+                                var option = document.createElement("option");
+                                option.setAttribute("value", i);
+                                option.text = applicationContext.tasks[i].uid;
+                                selectList.appendChild(option);
+                            }
+                        }    
```

Comment:
it's quite important to separate separable parts of logic to functions or even objects. Such separation increases readability of code and allows us to reuse this code in future. So I would suggest separation of code that adds `select` element to separate function:

```
success: function (data) {
  function populateTasksDropdown(tasks): {
    var selectList = document.createElement("select");
    selectList.setAttribute("id", "tasks");
    selectList.setAttribute("name", "tasks");
    choiceGraphic.appendChild(selectList);
      for (var i = 0; i < applicationContext.tasks.length; i++) {
        var option = document.createElement("option");
        option.setAttribute("value", i);
        option.text = applicationContext.tasks[i].uid;
        selectList.appendChild(option);
    }
  }

  applicationContext.tasks = data.tasks;
  populateTasksDropdown(data.tasks)
}
```

In such way we separated logic for `select` adding and added the name for this function. So now developers do not need to read all code to understand what it does.

Reply:
To improve readability of the code, the recommended function has been created.

Reply:
Input parameter of the created function is not used. My example also has this mistake. It's better to use function parameters instead of global variables since in such way we create more reusable function. So we need:

```
success: function (data) {
  function populateTasksDropdown(tasks): {
    var selectList = document.createElement("select");
    selectList.setAttribute("id", "tasks");
    selectList.setAttribute("name", "tasks");
    choiceGraphic.appendChild(selectList);
      for (var i = 0; i < tasks.length; i++) {
        var option = document.createElement("option");
        option.setAttribute("value", i);
        option.text = tasks[i].uid;
        selectList.appendChild(option);
    }
  }

  applicationContext.tasks = data.tasks;
  populateTasksDropdown(data.tasks)
}
```

So we now don't use global variable `applicationContext` in the body of `populateTasksDropdown`.

Reply:
Modified populateTasksDropdown function.